crates/libduckdb-sys/build.rs

```diff
@@ -160,7 +160,8 @@ mod build_bundled {
             .flag_if_supported("-stdlib=libstdc++")
             .flag_if_supported("/bigobj")
             .warnings(false)
-            .flag_if_supported("-w");
+            .flag_if_supported("-w")
```

Comment:
@Mytherin would it be better to have the preprocessor definition under win_target alone after line 167 like below instead:
cfg.define("_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR", None);

Reply:
@Mytherin It looks like we might need to pass the pre-processor definition for bindgen builder as well as it seem to use clang for building it at line 324.

```
        bindgen::builder()
            .trust_clang_mangling(false)
            .clang_arg("/D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR=1")
            .header(header.clone())
            .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
            .generate()
            .unwrap_or_else(|_| panic!("could not run bindgen on header {header}"))
            .write(Box::new(&mut output))
            .expect("could not write output of bindgen");
```

Reply:
@Mytherin It looks like latest version of runner seem to resolve the issue that we observe in Windows runner https://github.com/actions/runner-images/issues/10004#issuecomment-2177737349.

We could remove our patch from build.rs and re-try the pipeline.